Makefile.am

```diff
@@ -14,16 +14,49 @@ if ENABLE_DOCS
 SUBDIRS += doc
 endif
 
-if ENABLE_V4L2
-SUBDIRS += v4l2
-endif
-
 if ENABLE_CAPI
 SUBDIRS += capi
 endif
 
+#generate libyami.la for ENABLE_TESTS and ENABLE_V4L2
+SUBDIRS += .
+
+if ENABLE_V4L2
+SUBDIRS += v4l2
+endif
+
 if ENABLE_TESTS
 SUBDIRS += tests
 SUBDIRS += testscripts
 SUBDIRS += examples
 endif
+
+lib_LTLIBRARIES = libyami.la
+libyami_la_SOURCES =
```

Comment:
Please add `nodist_EXTRA_libyami_la_SOURCES = dummy.cpp` to force C++ linking...

https://www.gnu.org/software/automake/manual/html_node/Libtool-Convenience-Libraries.html